src/main/java/com/fibermc/essentialcommands/teleportation/PlayerTeleporter.java

```diff
@@ -72,8 +69,14 @@ private static void execTeleport(ServerPlayerEntity playerEntity, MinecraftLocat
         playerAccess.ec$getPlayerData().sendMessage(
             "teleport.done",
             playerProfile.shouldPrintTeleportCoordinates()
-                ? dest.toText(playerProfile)
-                : Text.literal("destination").setStyle(playerProfile.getStyle(TextFormatType.Default))
+                ? TextUtil.join(
+                    new Text[]{
+                        destName,
+                        dest.toText(playerProfile)
+                    },
+                    Text.literal(" ")
+                )
+                : destName
```

Comment:
This is a good behavioral change (always getting the `destName`, if it is available). Good catch.

The fact that this name building had to be moved out of `NamedMinecraftLocation` makes me think that the `[Some]MinecraftLocation` classes I've built are a little too clumsy when it comes to `Text` conversion. Might revisit this soon.

Either way, this behavior is better. Approved!